emloop/cli/common.py

```diff
@@ -213,6 +213,26 @@ def create_hooks(config: dict, model: AbstractModel,
     return hooks
 
 
+def create_main_loop(config: dict, model: Optional[AbstractModel]=None, dataset: Optional[AbstractDataset]=None,
+                     hooks: Optional[List[AbstractHook]]=None, eval: Optional[str]=None):
+    """
+    Create a main loop object.
+
+    :param config: config dict with main loop config
+    :param dataset: dataset object implementing the :py:class:`emloop.datasets.AbstractDataset` concept
+    :param model: model object implementing the :py:class:`emloop.models.AbstractModel` concept
+    :param hooks: list of hooks implementing the :py:class:`emloop.models.AbstractHook` concept
+
+    :return: main loop object
+    """
+
+    logging.info('Creating main loop')
+    kwargs = config['main_loop'] if 'main_loop' in config else {}
+    if eval is not None:
+        kwargs['extra_streams'] = []
```

Comment:
This will mutate the `extra_streams` item in the original dict, have you considered making a deep copy instead?